### PR TITLE
Add small Option usage refactorings

### PR DESCRIPTION
**Pull request by tranzystorek-io**

### CODE REVIEW

**Comment by erikh** [comment]
Thanks for this, I'll review it later today or tomorrow evening at the latest.

**Comment by erikh** [comment]
wow, I totally lost track of this.

File: tests/service/context.rs [hunk 1]
```diff
@@ -18,8 +18,8 @@ pub struct TestContext {
 impl TestContext {
     pub fn get_member(&mut self, network_id: String) -> Member {
         let mut member = Member::new(network_id, self.identity.clone());
-        if self.member_config.is_some() {
-            member.config = self.member_config.clone();
+        if let cfg @ Some(_) = self.member_config.clone() {
```
Comment (erikh) [inline tests/service/context.rs:21]: I've never seen this before. Can you explain it?

**Review by erikh — COMMENTED** [review]
I'd just like to know what that stanza means. No objection to it as of yet.

File: tests/service/context.rs (hunk 1 above)
Comment (tranzystorek-io) [reply tests/service/context.rs:21]: This just binds the whole `Option` object to the `cfg` variable, but only if it's the `Some` variant

Comment (tranzystorek-io) [reply tests/service/context.rs:21]: I guess this part can be off-putting to someone who sees it for the first time. If you don't like it we can revert it.

Comment (erikh) [reply tests/service/context.rs:21]: well I guess what's the difference here between this and:

```rust
if let Some(cfg) = self.member_config.clone() 
```

Comment (tranzystorek-io) [reply tests/service/context.rs:21]: > well I guess what's the difference here between this and:
> 
> ```rust
> if let Some(cfg) = self.member_config.clone() 
> ```

here you bind the `cfg` to the thing inside the option, so then if you want to assign it to the member.config it ends up a bit more verbose: `member.config = Some(cfg);`

Comment (erikh) [reply tests/service/context.rs:21]: ahh. I think I see now. Seems a little extra but sure, ok.

**Pull request merged**